src/index.tsx

```diff
@@ -196,18 +196,18 @@ export function useWebRTC() {
  * This hook provides live updates of room endpoints.
  * @returns An array of room endpoints.
  */
-export function useEndpoints() {
-  const [endpoints, setEndpoints] = useState<Endpoint[]>([]);
+export function useEndpoints<T extends Metadata, S extends Metadata>() {
```

Comment:
can't T and S be more descriptive? I don't know what do they mean

Reply:
also probably the metadata can be different for different tracks

for example, an audio track can have 'muted' property, which doesn't make sense for camera track